sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundleClassDiscriminatorIT.java

```diff
@@ -0,0 +1,324 @@
+
+    var foundPetOwner = repository.findById(savedOwner.getId());
+    assertThat(foundPetOwner).isPresent();
+    assertThat(foundPetOwner.get().getEntityClass()).isNull();
+    assertThat(foundPetOwner.get().getPet()).isInstanceOf(Cat.class);
+    assertThat(foundPetOwner.get().getPet().getEntityClass()).isEqualTo("cat");
+  }
+
+  private PetOwner createDogPetOwner() {
+    var petOwner = new PetOwner();
+    petOwner.setPet(createDog());
+    return petOwner;
+  }
+
+  private PetOwner createCatPetOwner() {
+    var petOwner = new PetOwner();
+    petOwner.setPet(createCat());
+    return petOwner;
+  }
+
+  private Dog createDog() {
+    Dog dog = new Dog();
+    dog.setName("Brutus");
+    dog.setBreed("Doberman");
+    dog.setWeight(25.4D);
+    dog.setColor("Black");
+    return dog;
+  }
+
+  private Cat createCat() {
+    Cat cat = new Cat();
+    cat.setName("Brutus");
+    cat.setAge(2);
+    cat.setWeight(25.4D);
+    cat.setColor("Black");
+    return cat;
+  }
+
+  private Person createPerson() {
+    PhoneNumber phoneNumber = new PhoneNumber().setNumber("+49123456789");
+    return new Person()
+        .setAge(18)
+        .setName("Robert")
+        .setBirthday(LocalDate.now().minusYears(44))
+        .setPhoneNumber(phoneNumber);
+  }
+
+  public static class WithClassDiscriminatorApp extends Application<MyConfiguration> {
+
+    private final SpringDataMongoBundle<MyConfiguration> springDataMongoBundle =
+        SpringDataMongoBundle.builder()
+            .withConfigurationProvider(MyConfiguration::getSpringDataMongo)
+            .withEntities(Dog.class, Cat.class)
+            .build();
+
+    private PersonRepository personRepository;
+    private PetOwnerRepository petOwnerRepository;
+
+    @Override
+    public void initialize(Bootstrap<MyConfiguration> bootstrap) {
+      bootstrap.addBundle(springDataMongoBundle);
+    }
+
+    @Override
+    public void run(MyConfiguration configuration, Environment environment) {
+      this.personRepository = springDataMongoBundle.createRepository(PersonRepository.class);
+      this.petOwnerRepository = springDataMongoBundle.createRepository(PetOwnerRepository.class);
+    }
+
+    public MongoOperations getMongoOperations() {
+      return springDataMongoBundle.getMongoOperations();
+    }
+
+    public PersonRepository getPersonRepository() {
+      return personRepository;
+    }
+
+    public PetOwnerRepository getPetOwnerRepository() {
+      return this.petOwnerRepository;
+    }
+  }
+
+  public static class WithoutClassDiscriminatorApp extends Application<MyConfiguration> {
+    private final SpringDataMongoBundle<MyConfiguration> springDataMongoBundle =
+        SpringDataMongoBundle.builder()
+            .withConfigurationProvider(MyConfiguration::getSpringDataMongo)
+            .withEntities(Dog.class, Cat.class)
+            .withoutClassDiscriminator(
+                new String[] {"org.sdase.commons.server.spring.data.mongo.example.model"})
```

Comment:
To me it feels very hacky to supply a String[] of packages. If it's not avoidable, couldn’t we simply use the applications base package or the package of the entities from `withEntities`? The loss of performance should be tolerable.